pkg/runner/cluster_swarm.go

```diff
@@ -162,8 +162,14 @@ func (*ClusterSwarmRunner) Run(input *api.RunInput) (*api.RunOutput, error) {
 			// if we are keeping the service, we must also keep the network.
 			return
 		}
-		if err := cli.NetworkRemove(ctx, networkID); err != nil {
-			log.Errorw("removing network", "name", sname)
+		err = retry(5, 1*time.Second, func() error {
```

Comment:
Are we sure that this is fundamentally different than setting a context deadline of 30 seconds and doing a single call? Is there a bug in Docker or something?

Reply:
I think `docker` was returning from the previous command while the `network` was still in use, so I think this is different to setting ctx deadline to 30sec. Then again it could have been triggered to the bug with the global ctx being picked up in the closure - I am not sure.

Will add a note to review once again, and will remove the `retry` in future PR if it is indeed not needed.